core/src/main/java/io/questdb/cairo/SymbolMapWriter.java

```diff
@@ -187,6 +187,14 @@ public void close() {
         nullValue = false;
     }
 
+    public void copySymbols(SymbolMapWriter another) {
+        final int symbolCount = another.getSymbolCount();
+        for (int i = 0; i < symbolCount; i++) {
+            put(another.valueOf(i));
```

Comment:
this is very naive and inefficient copy. The capacity problem stems from the reverse index of a symbol. This index has to be rebuilt. Index file is append-only, it doesn't have to be re-built and internal cache does not need to be re-populated.

Copy can:
- share the cache (map)
- share the append-only structures
- rebuild reverse index

Reply:
I understand symbol cache, it can be simply re-used.

What are the append-only structures you refer to? Is it `.o` and `.c` files? Do you want to hard link them instead of copying them?

Reply:
yes, we should:
- keep heap cache - it is already right size etc
- keep the .o and .c files - this is just append only "string" column
- perhaps keep the index writer object entirely
- create new index, this index uses "maxHash", derived from symbol capacity to bound hash of the "symbol" values. "maxHash" has to be updated and index has to be rebuilt. This index keys the "has" and "offset". Where "offset" is used to locate the "symbol" value.

Reply:
How about I re-use the cache, which is very easy to do, and have the rest of the changes go to another PR? This is not supposed to be a frequent operation to optimize it so heavily, and it will ease the anxiety of changing the symbol capacity for the users.

Reply:
lets do that

Reply:
I made the changes you requested now